MozillaFoundation/foundation.mozilla.org
# Icon SVG validation

# Description



Link to sample test page: https://foundation-s-icon-svg-v-yqnpyt.herokuapp.com/cms/snippets/nav/navmenu/edit/1/ (Credentials: admin2/admin2)
Related PRs/issues: #12232, [TP1-546](https://mozilla-hub.atlassian.net/browse/TP1-546)


This PR introduces the `SVGImageFormatValidator`, a new utility for validating SVG image formats. It is implemented in the icon fields of both the `NavFeaturedItemBlock` and `NavMenuFeaturedBlogTopicRelationship` models. Additionally, the factories for these models have been updated to use SVG images.

Lastly, this PR includes tests that validate the SVG format for both models, as well as direct tests for the `SVGImageFormatValidator` utility itself.

# Steps to test

To test the new SVG validation functionality, please follow these steps:

1. Access the Navigation Bar's edit page on the review app [here](https://foundation-s-icon-svg-v-yqnpyt.herokuapp.com/cms/snippets/nav/navmenu/edit/1/) using the credentials: admin2/admin2.

 2. Use Ctrl + F to locate "Test Featured Nav link 1", then proceed to add a new Featured Nav Link. Verify the validation functionality both without an image and with non-SVG format images.
 3. Navigate to the bottom of the page to the "Featured Blog Topics" section.
 4. Add a new Featured Topic and ensure the icon validation is correctly applied both with no image and with non-SVG format images.
 5. If they are both working as expected, testing is complete! 

# Screenshots:

**FeaturedNavItem (submitting with no image):**
<img width="500" alt="Screenshot 2024-04-24 at 9 04 47 PM" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/27ef812a-8281-4b5b-972e-19c2708d9760">


**FeaturedNavItem (submitting non-SVG image):**

<img width="500" alt="Screenshot 2024-04-24 at 9 05 02 PM" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/06cc92c0-0a80-4263-80bc-118edccfa3a4">


**NavMenuFeaturedBlogTopicRelationship (submitting with no image):**

<img width="500" alt="Screenshot 2024-04-24 at 9 05 31 PM" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/e6ef7b30-0da0-4b99-915d-61b600e2bede">

**NavMenuFeaturedBlogTopicRelationship (submitting non-SVG image):**

<img width="500" alt="Screenshot 2024-04-24 at 9 05 46 PM" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/67de2e61-0dec-41ec-b4a7-b97f96eb28d8">


┆Issue is synchronized with this [Jira Story](https://mozilla-hub.atlassian.net/browse/TP1-565)
